### PR TITLE
feat: add Linter element-templates linter plugin

### LINKED ISSUE #3357: Show element template errors in the error panel
### Problem you would like to solve

As a user I don't see element template errors in the `Problems` panel. As a result I cannot batch fix issues related to re-usable building blocks:

screely-1670929997045

### Proposed solution

As validation + the `Problems` panel is the defined way to check diagram consistency we should represent all element template errors in a diagram on the errors panel:

* Template not found errors
* Template outdated (warnings)
* Template property validation errors

### Alternatives considered

None.

### Breakdown

- [x] Move Element Templates into separate project
  - [x] C7 - decouple from internal Property definitions that are re-used in element-templates (e.g. https://github.com/bpmn-io/bpmn-js-properties-panel/blob/e1d2cdbe248da88f6f3a546d5e42917bb9dc83d7/src/provider/element-templates/properties/ErrorProperties.js#L47-L51)
- [x] Commit erroneous values to XML so linting can be done
  - [x] needs to be done in `@bpmn-io/properties-panel`: https://github.com/bpmn-io/properties-panel/pull/252
  - [x] Validate that ID checking still works: https://github.com/bpmn-io/bpmn-js-properties-panel/pull/938
- [x] Create linter plugin reusing error checking logic https://github.com/bpmn-io/bpmn-js-element-templates/pull/3
- [ ] Integrate Linter Plugin into Camunda Modeler: https://github.com/camunda/camunda-modeler/pull/3673
  - [ ] Needs to be configured with available element templates as no bpmn-js instance is available to the linter plugin
_No response_

### PR DESCRIPTION
**Pull request by marstamm**
Recording 2023-07-11 at 09 18 40
closes #3357

### CODE REVIEW

**Comment by marstamm** [comment]
Pinging @philippfromme as you already did the review on https://github.com/bpmn-io/bpmn-js-element-templates/pull/3

File: client/src/app/TabsProvider.js [hunk 1]
```diff
@@ -194,10 +197,16 @@ export default class TabsProvider {
             action: 'create-cloud-bpmn-diagram'
           } ];
         },
-        getLinter(plugins) {
+        async getLinter(plugins = [], tab, app) {
```
[the remark this answers is not in the document]
Comment (marstamm) [reply client/src/app/TabsProvider.js:200]: We could limit it to `getConfig` so we don't pass the whole app

[the remark this answers is not in the document]
Comment (marstamm) [reply client/src/app/TabsProvider.js]: cf. [ae885ef](https://github.com/camunda/camunda-modeler/pull/3673/commits/ae885ef9ada7d8c7558f51db662474d86b26fc8a)

**Review by philippfromme — APPROVED** [review]
✅

**Pull request merged**